### ISSUE TITLE
Pushing to pull requests results in unrendered HTML on Gitlab forges

### ISSUE THREAD

**Issue by Thaodan** [magit/forge#318]
When pushing a new commit to a repository there is a new message about the new commit eg in my latest case:

```
Thaodan 2 hours ago
added 1 commit

<ul><li>f3846f73 - [sb2] Refactor path handling. Contributes to
JB#51819</li></ul>
```

This that intended or a bug?
My forge version is: 20201201.1713

**Comment by tarsius** [comment]
> When pushing a new commit to a repository there is a new message about the new commit eg in my latest case:

Either I completely misunderstand what you are talking about or this has nothing to do with Forge. Forge has no say in what Gitlab.com does in response to you effectively running `git pull ...`. If this is actually what you are talking about, then explain why you think Forge has anything to do with it.

**Comment by Thaodan** [comment]
I meant something like that:

**Comment by Thaodan** [comment]
![Screenshot_20201209_003507](https://user-images.githubusercontent.com/1747359/101549498-76821600-39b6-11eb-8a0e-dffa9969f866.png)

**Comment by tarsius** [comment]
That's certainly unexpected and I've never seen it before. Does that happen just with the gitlab instance we talked about before or also on gitlab.com?

**Comment by Thaodan** [comment]
Yes same problem see my test PR on one of my projects:
https://gitlab.com/Thaodan/aur_helpers/-/merge_requests/1

**Comment by tarsius** [comment]
Actually I just remembered that is a feature. I didn't remember because I don't really use Forge with Gitlab. This was "implemented" in b019dfa6c9d3ff084b209837fce638bf4f117d5b.

**Comment by tarsius** [comment]
The `<ul><li>` part is new, I believe. Or maybe adding a commit did not result in the creation of what the Gitlab API calls a "note".

In any case this is rather noisy and we should probably filter out some of that text. But I am not thrilled about the prospect of having to use some heuristic again to guess the type of a "note" (see the comment removed by the mentioned commit). So I probably won't tackle this any time soon.

**Comment by tarsius** [comment]
Classifying as a "third-party bug" because they really should have a type field on the very generic "note" object type.

**Comment by Thaodan** [comment]
> The `<ul><li>` part is new, I believe. Or maybe adding a commit did not
> result in the creation of what the Gitlab API calls a "note".
> 
> In any case this is rather noisy and we should probably filter out some of
> that text. But I am not thrilled about the prospect of having to use some
> heuristic again to guess the type of a "note" (see the comment removed by
> the mentioned commit). So I probably won't tackle this any time soon.
>
I don't think filtering those out make sense as they come into context when 
something changes e.g.  a comment reacting to line of code that is no longer 
present.

**Comment by tarsius** [comment]
I agree, these "notes" are useful, but what would be even better is if we could easily tell them apart from comments that were written by human beings. We could then provide different functionality depending on the type of the "note" at point. For example, we could disallow editing "status updates". It is currently possible to *attempt* to do so; of course it eventually leads to an error, but unfortunately not upfront.

**Comment by Thaodan** [comment]
Reported the problem upstream: 
https://gitlab.com/gitlab-org/gitlab/-/issues/292422

**Comment by Thaodan** [comment]
Feel free to ad more if I had missed something.

**Comment by tarsius** [comment]
So far they have talked about who's responsibility it is to look into this. :roll_eyes:

**Comment by tarsius** [comment]
I'm closing this and instead adding the "once in a while revisit" label (`.`). Without upstream moving, there ain't much I can do.
